### ISSUE TITLE
Unable to import "pudl" library when installing powergenome

### ISSUE THREAD

**Issue by wehuang16** [PowerGenome/PowerGenome#282]
I followed the installation instructions on the readme file (https://github.com/PowerGenome/PowerGenome/blob/master/README.md) in order to install powergenome: using Mambaforge, downloading the PUDL database from Zenodo, downloading other data files, creating the “env” file, activating the powergenome environment, and etc.. However, when I try to run one of the Jupyter notebook “ipynb” files, such as 
“existing and new generators.ipynb” (https://github.com/PowerGenome/PowerGenome/blob/master/notebooks/Existing%20and%20new%20generators.ipynb). I got the error that “No module named ‘pudl’” (see image below).  I am using Windows 10. 

![Capture](https://github.com/PowerGenome/PowerGenome/assets/82864676/167111d6-4fdf-42f5-9d9f-c91055e0ac89)

I have tried to start over the powergenome installation process from the readme file several times. However, the issue of “No module named ‘pudl’” still persists. Does anyone know how I could resolve this issue of installing powergenome?

**Comment by gschivley** [comment]
Hi, sorry to hear that you're having issues. There are a few things that could be going on. When you run the notebook are you sure that it is using the correct kernel? 

Try activating the power genome environment (`conda activate powergenome`) and running `conda list`. This should list all the packages installed in the environment. Can you share it here?

**Comment by wehuang16** [comment]
Hey Greg, thanks for reaching back to me! Below is my "conda list":
![image_2023-09-19_074915944](https://github.com/PowerGenome/PowerGenome/assets/82864676/94b73850-9dfa-47e3-b6b9-588f29af1075)

**Comment by gschivley** [comment]
Looks like somehow the environment is not being set up correctly. See my full list of packages below for reference.

Are you installing with pip or git clone and then using mamba to create an environment with the `environment.yml` file?


```
❯ conda list
# packages in environment at /Users/.../miniconda3/envs/powergenome:
#
# Name                    Version                   Build  Channel
addfips                   0.4.0              pyhd8ed1ab_1    conda-forge
adjusttext                0.7.3.1                    py_1    conda-forge
affine                    2.4.0              pyhd8ed1ab_0    conda-forge
aiohttp                   3.7.4.post0     py310he2143c4_1    conda-forge
altair                    5.0.1              pyhd8ed1ab_1    conda-forge
altair-data-server        0.4.1                    pypi_0    pypi
altair-saver              0.5.0                    pypi_0    pypi
altair-viewer             0.4.0                    pypi_0    pypi
anyio                     3.7.0              pyhd8ed1ab_1    conda-forge
aom                       3.5.0                h7ea286d_0    conda-forge
appnope                   0.1.3              pyhd8ed1ab_0    conda-forge
arelle-release            2.2.4              pyhd8ed1ab_1    conda-forge
argon2-cffi               21.3.0             pyhd8ed1ab_0    conda-forge
argon2-cffi-bindings      21.2.0          py310h8e9501a_3    conda-forge
asttokens                 2.2.1              pyhd8ed1ab_0    conda-forge
async-lru                 2.0.2              pyhd8ed1ab_0    conda-forge
async-timeout             3.0.1                   py_1000    conda-forge
attrs                     23.1.0             pyh71513ae_1    conda-forge
aws-c-auth                0.6.27               h71e70db_1    conda-forge
aws-c-cal                 0.5.26               h5f09868_1    conda-forge
aws-c-common              0.8.19               hb547adb_0    conda-forge
aws-c-compression         0.2.16               hf5f6f59_7    conda-forge
aws-c-event-stream        0.2.20               h589842e_7    conda-forge
aws-c-http                0.7.7                h262875d_4    conda-forge
aws-c-io                  0.13.21              h6297769_5    conda-forge
aws-c-mqtt                0.8.11               h6d34d79_1    conda-forge
aws-c-s3                  0.3.0                h2ea4599_2    conda-forge
aws-c-sdkutils            0.1.9                hf5f6f59_2    conda-forge
aws-checksums             0.1.14               hf5f6f59_7    conda-forge
aws-crt-cpp               0.20.2               hcf593e6_0    conda-forge
aws-sdk-cpp               1.10.57             h015ea4b_13    conda-forge
babel                     2.12.1             pyhd8ed1ab_1    conda-forge
backcall                  0.2.0              pyh9f0ad1d_0    conda-forge
backports                 1.0                pyhd8ed1ab_3    conda-forge
backports.functools_lru_cache 1.6.4              pyhd8ed1ab_0    conda-forge
backports.zoneinfo        0.2.1           py310hbe9552e_7    conda-forge
beautifulsoup4            4.12.2             pyha770c72_0    conda-forge
black                     23.3.0          py310hbe9552e_1    conda-forge
bleach                    6.0.0              pyhd8ed1ab_0    conda-forge
blinker                   1.6.2              pyhd8ed1ab_0    conda-forge
blosc                     1.21.4               hc338f07_0    conda-forge
bokeh                     2.4.3              pyhd8ed1ab_3    conda-forge
boost-cpp                 1.78.0               hf1d6563_2    conda-forge
boto3                     1.26.146           pyhd8ed1ab_0    conda-forge
botocore                  1.29.146           pyhd8ed1ab_0    conda-forge
branca                    0.6.0              pyhd8ed1ab_0    conda-forge
brotli                    1.0.9                h1a8c8d9_8    conda-forge
brotli-bin                1.0.9                h1a8c8d9_8    conda-forge
brotlipy                  0.7.0           py310h8e9501a_1005    conda-forge
brunsli                   0.1                  h9f76cd9_0    conda-forge
build                     0.10.0                   pypi_0    pypi
bzip2                     1.0.8                h3422bc3_4    conda-forge
c-ares                    1.19.1               hb547adb_0    conda-forge
c-blosc2                  2.9.2                h068da5f_0    conda-forge
ca-certificates           2023.5.7             hf0a4a13_0    conda-forge
cached-property           1.5.2                hd8ed1ab_1    conda-forge
cached_property           1.5.2              pyha770c72_1    conda-forge
cachetools                5.3.0              pyhd8ed1ab_0    conda-forge
cairo                     1.16.0            h73a0509_1014    conda-forge
catalystcoop.dbfread      3.0.0                      py_0    conda-forge
catalystcoop.ferc-xbrl-extractor 0.6.1              pyhd8ed1ab_1    conda-forge
catalystcoop.pudl         2022.11.30         pyhd8ed1ab_0    conda-forge
cchardet                  2.1.7           py310h0f1eb42_4    conda-forge
certifi                   2023.5.7           pyhd8ed1ab_0    conda-forge
cffi                      1.15.1          py310h2399d43_3    conda-forge
cfgv                      3.3.1              pyhd8ed1ab_0    conda-forge
cfitsio                   4.2.0                h2f961c4_0    conda-forge
cftime                    1.6.2           py310hf1a086a_1    conda-forge
chardet                   4.0.0           py310hbe9552e_3    conda-forge
charls                    2.4.2                h13dd4ca_0    conda-forge
charset-normalizer        3.1.0              pyhd8ed1ab_0    conda-forge
click                     8.1.3           unix_pyhd8ed1ab_2    conda-forge
click-plugins             1.1.1                      py_0    conda-forge
cligj                     0.7.2              pyhd8ed1ab_1    conda-forge
cloudpickle               2.2.1              pyhd8ed1ab_0    conda-forge
colorama                  0.4.6              pyhd8ed1ab_0    conda-forge
coloredlogs               15.0.1             pyhd8ed1ab_3    conda-forge
comm                      0.1.3              pyhd8ed1ab_0    conda-forge
conda-souschef            2.2.3              pyhd8ed1ab_0    conda-forge
contourpy                 1.0.7           py310h2887b22_0    conda-forge
coverage                  7.2.7           py310h2aa6e3c_0    conda-forge
cryptography              41.0.1          py310hdd3b5e7_0    conda-forge
curl                      8.1.2                h912dcd9_0    conda-forge
cycler                    0.11.0             pyhd8ed1ab_0    conda-forge
cytoolz                   0.12.0          py310h8e9501a_1    conda-forge
dask                      2022.6.1           pyhd8ed1ab_0    conda-forge
dask-core                 2022.6.1           pyhd8ed1ab_0    conda-forge
dask-geopandas            0.3.1              pyhd8ed1ab_0    conda-forge
datapackage               1.15.2             pyh44b312d_0    conda-forge
dav1d                     1.2.1                hb547adb_0    conda-forge
debugpy                   1.6.7           py310h0f1eb42_0    conda-forge
decorator                 5.1.1              pyhd8ed1ab_0    conda-forge
defusedxml                0.7.1              pyhd8ed1ab_0    conda-forge
distlib                   0.3.6              pyhd8ed1ab_0    conda-forge
distributed               2022.6.1           pyhd8ed1ab_0    conda-forge
dnspython                 2.3.0              pyhd8ed1ab_0    conda-forge
docutils                  0.20.1                   pypi_0    pypi
email-validator           1.3.1              pyhd8ed1ab_0    conda-forge
entrypoints               0.4                pyhd8ed1ab_0    conda-forge
et_xmlfile                1.1.0              pyhd8ed1ab_0    conda-forge
exceptiongroup            1.1.1              pyhd8ed1ab_0    conda-forge
executing                 1.2.0              pyhd8ed1ab_0    conda-forge
expat                     2.5.0                hb7217d7_1    conda-forge
filelock                  3.12.0             pyhd8ed1ab_0    conda-forge
fiona                     1.9.3           py310h2e51ddd_0    conda-forge
flatten-dict              0.4.2              pyhd8ed1ab_1    conda-forge
flit-core                 3.9.0              pyhd8ed1ab_0    conda-forge
folium                    0.14.0             pyhd8ed1ab_0    conda-forge
font-ttf-dejavu-sans-mono 2.37                 hab24e00_0    conda-forge
font-ttf-inconsolata      3.000                h77eed37_0    conda-forge
font-ttf-source-code-pro  2.038                h77eed37_0    conda-forge
font-ttf-ubuntu           0.83                 hab24e00_0    conda-forge
fontconfig                2.14.2               h82840c6_0    conda-forge
fonts-conda-ecosystem     1                             0    conda-forge
fonts-conda-forge         1                             0    conda-forge
fonttools                 4.39.4          py310h2aa6e3c_0    conda-forge
freetype                  2.12.1               hd633e50_1    conda-forge
freexl                    1.0.6                h1a8c8d9_1    conda-forge
frictionless              4.40.8             pyh6c4a22f_0    conda-forge
frozendict                2.3.8           py310h2aa6e3c_0    conda-forge
fsspec                    2022.11.0          pyhd8ed1ab_0    conda-forge
gcsfs                     2022.11.0          pyhd8ed1ab_0    conda-forge
gdal                      3.6.3           py310h0b34360_0    conda-forge
geopandas                 0.12.2             pyhd8ed1ab_0    conda-forge
geopandas-base            0.12.2             pyha770c72_0    conda-forge
geos                      3.11.1               hb7217d7_0    conda-forge
geotiff                   1.7.1                h435296a_7    conda-forge
gettext                   0.21.1               h0186832_0    conda-forge
gflags                    2.2.2             hc88da5d_1004    conda-forge
giflib                    5.2.1                h1a8c8d9_3    conda-forge
glog                      0.6.0                h6da1cb0_0    conda-forge
google-api-core           2.11.0             pyhd8ed1ab_0    conda-forge
google-auth               2.19.1             pyh1a96a4e_0    conda-forge
google-auth-oauthlib      1.0.0              pyhd8ed1ab_0    conda-forge
google-cloud-core         2.3.2              pyhd8ed1ab_0    conda-forge
google-cloud-storage      2.9.0              pyh1a96a4e_0    conda-forge
google-crc32c             1.1.2           py310he58995c_4    conda-forge
google-resumable-media    2.5.0              pyhd8ed1ab_0    conda-forge
googleapis-common-protos  1.57.1             pyhd8ed1ab_0    conda-forge
grayskull                 2.3.0              pyhd8ed1ab_1    conda-forge
greenlet                  2.0.2           py310h1253130_1    conda-forge
grpcio                    1.54.2          py310h1253130_2    conda-forge
h11                       0.14.0                   pypi_0    pypi
h3-py                     3.7.6           py310h0f1eb42_0    conda-forge
h5py                      3.9.0                    pypi_0    pypi
hdf4                      4.2.15               h8111dcc_6    conda-forge
hdf5                      1.12.2          nompi_ha7af310_101    conda-forge
humanfriendly             10.0            py310hbe9552e_4    conda-forge
hypothesis                6.76.0             pyha770c72_0    conda-forge
icu                       70.1                 h6b3803e_0    conda-forge
identify                  2.5.24             pyhd8ed1ab_0    conda-forge
idna                      3.4                pyhd8ed1ab_0    conda-forge
ijson                     3.2.0.post0        pyhd8ed1ab_0    conda-forge
imagecodecs               2023.1.23       py310h2d7c47d_1    conda-forge
imageio                   2.28.1             pyh24c5eb1_0    conda-forge
importlib-metadata        6.6.0              pyha770c72_0    conda-forge
importlib_metadata        6.6.0                hd8ed1ab_0    conda-forge
importlib_resources       5.12.0             pyhd8ed1ab_0    conda-forge
iniconfig                 2.0.0              pyhd8ed1ab_0    conda-forge
ipykernel                 6.23.1             pyh736e0ef_0    conda-forge
ipython                   8.14.0             pyhd1c38e8_0    conda-forge
isodate                   0.6.1              pyhd8ed1ab_0    conda-forge
isort                     5.12.0             pyhd8ed1ab_1    conda-forge
jaraco-classes            3.2.3                    pypi_0    pypi
jedi                      0.18.2             pyhd8ed1ab_0    conda-forge
jinja2                    3.1.2              pyhd8ed1ab_1    conda-forge
jmespath                  1.0.1              pyhd8ed1ab_0    conda-forge
joblib                    1.2.0              pyhd8ed1ab_0    conda-forge
json-c                    0.16                 hc449e50_0    conda-forge
json5                     0.9.5              pyh9f0ad1d_0    conda-forge
jsonlines                 3.1.0              pyhd8ed1ab_0    conda-forge
jsonpointer               2.0                        py_0    conda-forge
jsonschema                4.17.3             pyhd8ed1ab_0    conda-forge
jupyter-lsp               2.2.0              pyhd8ed1ab_0    conda-forge
jupyter_client            8.2.0              pyhd8ed1ab_0    conda-forge
jupyter_core              5.3.0           py310hbe9552e_0    conda-forge
jupyter_events            0.6.3              pyhd8ed1ab_0    conda-forge
jupyter_server            2.6.0              pyhd8ed1ab_0    conda-forge
jupyter_server_terminals  0.4.4              pyhd8ed1ab_1    conda-forge
jupyterlab                4.0.1              pyhd8ed1ab_0    conda-forge
jupyterlab_pygments       0.2.2              pyhd8ed1ab_0    conda-forge
jupyterlab_server         2.22.1             pyhd8ed1ab_0    conda-forge
jxrlib                    1.1                  h27ca646_2    conda-forge
kealib                    1.5.0                hfd766a6_0    conda-forge
keyring                   23.13.1                  pypi_0    pypi
kiwisolver                1.4.4           py310h2887b22_1    conda-forge
krb5                      1.20.1               h69eda48_0    conda-forge
lazy_loader               0.2                pyhd8ed1ab_0    conda-forge
lcms2                     2.15                 hd835a16_1    conda-forge
lerc                      4.0.0                h9a09cb3_0    conda-forge
libabseil                 20230125.2      cxx17_h13dd4ca_2    conda-forge
libaec                    1.0.6                hb7217d7_1    conda-forge
libarrow                  10.0.1          he2904f1_27_cpu    conda-forge
libavif                   0.11.1               h9f83d30_2    conda-forge
libblas                   3.9.0           17_osxarm64_openblas    conda-forge
libbrotlicommon           1.0.9                h1a8c8d9_8    conda-forge
libbrotlidec              1.0.9                h1a8c8d9_8    conda-forge
libbrotlienc              1.0.9                h1a8c8d9_8    conda-forge
libcblas                  3.9.0           17_osxarm64_openblas    conda-forge
libcrc32c                 1.1.2                hbdafb3b_0    conda-forge
libcurl                   8.1.2                h912dcd9_0    conda-forge
libcxx                    16.0.5               h4653b0c_0    conda-forge
libdeflate                1.17                 h1a8c8d9_0    conda-forge
libedit                   3.1.20191231         hc8eb9b7_2    conda-forge
libev                     4.33                 h642e427_1    conda-forge
libevent                  2.1.12               h2757513_1    conda-forge
libexpat                  2.5.0                hb7217d7_1    conda-forge
libffi                    3.4.2                h3422bc3_5    conda-forge
libgdal                   3.6.3                h2c41457_0    conda-forge
libgfortran               5.0.0           12_2_0_hd922786_31    conda-forge
libgfortran5              12.2.0              h0eea778_31    conda-forge
libglib                   2.76.3               h24e9cb9_0    conda-forge
libgoogle-cloud           2.11.0               he22f4c0_0    conda-forge
libgrpc                   1.54.2               h0a338ca_2    conda-forge
libiconv                  1.17                 he4db4b2_0    conda-forge
libjpeg-turbo             2.1.5.1              h1a8c8d9_0    conda-forge
libkml                    1.3.0             h41464e4_1015    conda-forge
liblapack                 3.9.0           17_osxarm64_openblas    conda-forge
libllvm14                 14.0.6               hd1a9a77_3    conda-forge
libnetcdf                 4.9.1           nompi_hda66dd9_102    conda-forge
libnghttp2                1.52.0               hae82a92_0    conda-forge
libopenblas               0.3.23          openmp_hc731615_0    conda-forge
libpng                    1.6.39               h76d750c_0    conda-forge
libpq                     15.2                 h1a28acd_0    conda-forge
libprotobuf               3.21.12              hb5ab8b9_0    conda-forge
librttopo                 1.1.0               h844f84d_12    conda-forge
libsodium                 1.0.18               h27ca646_1    conda-forge
libspatialindex           1.9.3                hbdafb3b_4    conda-forge
libspatialite             5.0.1               h14115fc_23    conda-forge
libsqlite                 3.42.0               hb31c410_0    conda-forge
libssh2                   1.11.0               h7a5bd25_0    conda-forge
libthrift                 0.18.1               ha061701_2    conda-forge
libtiff                   4.5.0                hdc14d85_5    conda-forge
libutf8proc               2.8.0                h1a8c8d9_0    conda-forge
libwebp-base              1.3.0                h1a8c8d9_0    conda-forge
libxcb                    1.15                 hf346824_0    conda-forge
libxml2                   2.10.3               h67585b2_4    conda-forge
libxslt                   1.1.37               h1bd8bc4_0    conda-forge
libzip                    1.9.2                h76ab92c_1    conda-forge
libzlib                   1.2.13               h03a7124_4    conda-forge
libzopfli                 1.0.3                h9f76cd9_0    conda-forge
linear-tsv                1.1.0                      py_1    conda-forge
llvm-openmp               16.0.5               h1c12783_0    conda-forge
llvmlite                  0.40.0          py310h95b248a_0    conda-forge
locket                    1.0.0              pyhd8ed1ab_0    conda-forge
lxml                      4.9.2           py310h85b680a_0    conda-forge
lz4                       4.3.2           py310ha6df754_0    conda-forge
lz4-c                     1.9.4                hb7217d7_0    conda-forge
mapclassify               2.5.0              pyhd8ed1ab_1    conda-forge
markdown-it-py            2.2.0              pyhd8ed1ab_0    conda-forge
marko                     1.3.0              pyhd8ed1ab_0    conda-forge
markupsafe                2.1.3           py310h2aa6e3c_0    conda-forge
matplotlib-base           3.6.3           py310h78c5c2f_0    conda-forge
matplotlib-inline         0.1.6              pyhd8ed1ab_0    conda-forge
mdurl                     0.1.0              pyhd8ed1ab_0    conda-forge
mistune                   2.0.5              pyhd8ed1ab_0    conda-forge
more-itertools            9.1.0                    pypi_0    pypi
msgpack-python            1.0.5           py310h2887b22_0    conda-forge
multidict                 6.0.4           py310h8e9501a_0    conda-forge
munch                     3.0.0              pyhd8ed1ab_0    conda-forge
munkres                   1.1.4              pyh9f0ad1d_0    conda-forge
mypy_extensions           1.0.0              pyha770c72_0    conda-forge
nb_conda_kernels          2.3.1           py310hbe9552e_2    conda-forge
nbclient                  0.8.0              pyhd8ed1ab_0    conda-forge
nbconvert-core            7.4.0              pyhd8ed1ab_0    conda-forge
nbformat                  5.9.0              pyhd8ed1ab_0    conda-forge
ncurses                   6.3                  h07bb92c_1    conda-forge
nest-asyncio              1.5.6              pyhd8ed1ab_0    conda-forge
netcdf4                   1.6.3           nompi_py310h896f513_100    conda-forge
networkx                  2.8.8              pyhd8ed1ab_0    conda-forge
nodeenv                   1.8.0              pyhd8ed1ab_0    conda-forge
notebook-shim             0.2.3              pyhd8ed1ab_0    conda-forge
nspr                      4.35                 hb7217d7_0    conda-forge
nss                       3.89                 h789eff7_0    conda-forge
numba                     0.57.0          py310hb9b3264_1    conda-forge
numpy                     1.23.5          py310h5d7c261_0    conda-forge
oauthlib                  3.2.2              pyhd8ed1ab_0    conda-forge
openjpeg                  2.5.0                hbc2ba62_2    conda-forge
openpyxl                  3.1.2           py310h2aa6e3c_0    conda-forge
openssl                   3.1.1                h53f4e23_1    conda-forge
orc                       1.8.3                ha98e9e8_1    conda-forge
outcome                   1.2.0                    pypi_0    pypi
overrides                 7.3.1              pyhd8ed1ab_0    conda-forge
packaging                 23.1               pyhd8ed1ab_0    conda-forge
pandas                    1.4.4           py310h2b830bf_0    conda-forge
pandocfilters             1.5.0              pyhd8ed1ab_0    conda-forge
parso                     0.8.3              pyhd8ed1ab_0    conda-forge
partd                     1.4.0              pyhd8ed1ab_0    conda-forge
pathlib2                  2.3.7.post1     py310hbe9552e_2    conda-forge
pathspec                  0.11.1             pyhd8ed1ab_0    conda-forge
patsy                     0.5.3              pyhd8ed1ab_0    conda-forge
pcre2                     10.40                hb34f9b4_0    conda-forge
petl                      1.7.4              pyh44b312d_0    conda-forge
pexpect                   4.8.0              pyh1a96a4e_2    conda-forge
pickleshare               0.7.5                   py_1003    conda-forge
pillow                    9.5.0           py310h60ecbdf_1    conda-forge
pip                       23.1.2             pyhd8ed1ab_0    conda-forge
pixman                    0.40.0               h27ca646_0    conda-forge
pkginfo                   1.9.6              pyhd8ed1ab_0    conda-forge
pkgutil-resolve-name      1.3.10             pyhd8ed1ab_0    conda-forge
platformdirs              3.5.1              pyhd8ed1ab_0    conda-forge
pluggy                    1.0.0              pyhd8ed1ab_5    conda-forge
poppler                   23.03.0              h15f6d88_1    conda-forge
poppler-data              0.4.12               hd8ed1ab_0    conda-forge
portpicker                1.6.0                    pypi_0    pypi
postgresql                15.2                 h45c140d_0    conda-forge
powergenome               0.6.1                     dev_0    <develop>
pre-commit                3.3.2              pyha770c72_0    conda-forge
progressbar2              4.2.0              pyhd8ed1ab_0    conda-forge
proj                      9.1.1                h13f728c_2    conda-forge
prometheus_client         0.17.0             pyhd8ed1ab_0    conda-forge
prompt-toolkit            3.0.38             pyha770c72_0    conda-forge
prompt_toolkit            3.0.38               hd8ed1ab_0    conda-forge
protobuf                  4.21.12         py310h0f1eb42_0    conda-forge
psutil                    5.9.5           py310h8e9501a_0    conda-forge
pthread-stubs             0.4               h27ca646_1001    conda-forge
ptyprocess                0.7.0              pyhd3deb0d_0    conda-forge
pure_eval                 0.2.2              pyhd8ed1ab_0    conda-forge
pvlib                     0.10.1                   pypi_0    pypi
pyarrow                   10.0.1          py310h89f3c6b_27_cpu    conda-forge
pyasn1                    0.4.8                      py_0    conda-forge
pyasn1-modules            0.2.7                      py_0    conda-forge
pycparser                 2.21               pyhd8ed1ab_0    conda-forge
pydantic                  1.10.8          py310h2aa6e3c_0    conda-forge
pygeos                    0.13            py310h9356385_2    conda-forge
pygments                  2.15.1             pyhd8ed1ab_0    conda-forge
pyjwt                     2.7.0              pyhd8ed1ab_0    conda-forge
pyobjc-core               9.1.1           py310h44ed3dd_0    conda-forge
pyobjc-framework-cocoa    9.1.1           py310h44ed3dd_0    conda-forge
pyogrio                   0.5.1           py310h2e51ddd_0    conda-forge
pyopenssl                 23.2.0             pyhd8ed1ab_1    conda-forge
pyparsing                 3.0.9              pyhd8ed1ab_0    conda-forge
pyproj                    3.4.1           py310h9862ddf_1    conda-forge
pyproject-hooks           1.0.0                    pypi_0    pypi
pyrsistent                0.19.3          py310h8e9501a_0    conda-forge
pysocks                   1.7.1              pyha2e5f31_6    conda-forge
pytest                    7.3.1              pyhd8ed1ab_0    conda-forge
pytest-cov                4.1.0              pyhd8ed1ab_0    conda-forge
python                    3.10.11         h3ba56d0_0_cpython    conda-forge
python-dateutil           2.8.2              pyhd8ed1ab_0    conda-forge
python-dotenv             1.0.0              pyhd8ed1ab_0    conda-forge
python-fastjsonschema     2.17.1             pyhd8ed1ab_0    conda-forge
python-json-logger        2.0.7              pyhd8ed1ab_0    conda-forge
python-slugify            8.0.1              pyhd8ed1ab_0    conda-forge
python-snappy             0.6.1           py310h7f31a72_0    conda-forge
python-utils              3.6.0              pyhd8ed1ab_0    conda-forge
python_abi                3.10                    3_cp310    conda-forge
pytz                      2023.3             pyhd8ed1ab_0    conda-forge
pyu2f                     0.1.5              pyhd8ed1ab_0    conda-forge
pywavelets                1.4.1           py310hf1a086a_0    conda-forge
pyyaml                    6.0             py310h8e9501a_5    conda-forge
pyzmq                     25.1.0          py310h30b7201_0    conda-forge
rapidfuzz                 3.1.1           py310h1253130_0    conda-forge
rasterio                  1.3.6           py310h2630d6e_0    conda-forge
re2                       2023.03.02           hc5e2d97_0    conda-forge
readline                  8.2                  h92ec313_1    conda-forge
readme-renderer           37.3                     pypi_0    pypi
regex                     2023.6.3        py310h2aa6e3c_0    conda-forge
requests                  2.31.0             pyhd8ed1ab_0    conda-forge
requests-oauthlib         1.3.1              pyhd8ed1ab_0    conda-forge
requests-toolbelt         1.0.0                    pypi_0    pypi
rfc3339-validator         0.1.4              pyhd8ed1ab_0    conda-forge
rfc3986                   2.0.0              pyhd8ed1ab_0    conda-forge
rfc3986-validator         0.1.1              pyh9f0ad1d_0    conda-forge
rich                      13.4.1             pyhd8ed1ab_0    conda-forge
rsa                       4.9                pyhd8ed1ab_0    conda-forge
rtree                     1.0.1           py310ha3239f5_1    conda-forge
ruamel.yaml               0.17.31         py310h2aa6e3c_0    conda-forge
ruamel.yaml.clib          0.2.7           py310h8e9501a_1    conda-forge
ruamel.yaml.jinja2        0.2.4                      py_1    conda-forge
s3transfer                0.6.1              pyhd8ed1ab_0    conda-forge
scikit-image              0.20.0          py310h2b830bf_1    conda-forge
scikit-learn              1.1.3           py310ha00a7cd_1    conda-forge
scipy                     1.9.3           py310ha0d8a01_2    conda-forge
seaborn                   0.12.2               hd8ed1ab_0    conda-forge
seaborn-base              0.12.2             pyhd8ed1ab_0    conda-forge
selenium                  4.11.2                   pypi_0    pypi
semver                    2.13.0             pyh9f0ad1d_0    conda-forge
send2trash                1.8.2              pyhd1c38e8_0    conda-forge
setuptools                67.7.2             pyhd8ed1ab_0    conda-forge
shapely                   2.0.1           py310h9356385_0    conda-forge
shellingham               1.5.1              pyhd8ed1ab_0    conda-forge
simpleeval                0.9.13          py310h8e9501a_0    conda-forge
six                       1.16.0             pyh6c4a22f_0    conda-forge
snappy                    1.1.10               h17c5cce_0    conda-forge
sniffio                   1.3.0              pyhd8ed1ab_0    conda-forge
snuggs                    1.4.7                      py_0    conda-forge
sortedcontainers          2.4.0              pyhd8ed1ab_0    conda-forge
soupsieve                 2.3.2.post1        pyhd8ed1ab_0    conda-forge
sqlalchemy                1.4.44          py310h8e9501a_0    conda-forge
sqlite                    3.42.0               h203b68d_0    conda-forge
stack_data                0.6.2              pyhd8ed1ab_0    conda-forge
statsmodels               0.14.0          py310ha11ecec_1    conda-forge
stdlib-list               0.8.0              pyhd8ed1ab_0    conda-forge
stringcase                1.2.0                      py_0    conda-forge
tableschema               1.19.3             pyh9f0ad1d_0    conda-forge
tabulate                  0.9.0              pyhd8ed1ab_1    conda-forge
tabulator                 1.53.5             pyhd8ed1ab_0    conda-forge
tblib                     1.7.0              pyhd8ed1ab_0    conda-forge
terminado                 0.17.1             pyhd1c38e8_0    conda-forge
text-unidecode            1.3                        py_0    conda-forge
threadpoolctl             3.1.0              pyh8a188c0_0    conda-forge
tifffile                  2023.4.12          pyhd8ed1ab_0    conda-forge
tiledb                    2.13.2               h9bd36d0_0    conda-forge
timezonefinder            6.0.2              pyhd8ed1ab_0    conda-forge
tinycss2                  1.2.1              pyhd8ed1ab_0    conda-forge
tk                        8.6.12               he1e0b03_0    conda-forge
toml                      0.10.2             pyhd8ed1ab_0    conda-forge
tomli                     2.0.1              pyhd8ed1ab_0    conda-forge
tomli-w                   1.0.0              pyhd8ed1ab_0    conda-forge
toolz                     0.12.0             pyhd8ed1ab_0    conda-forge
tornado                   6.3.2           py310h2aa6e3c_0    conda-forge
tqdm                      4.65.0             pyhd8ed1ab_1    conda-forge
traitlets                 5.9.0              pyhd8ed1ab_0    conda-forge
trio                      0.22.2                   pypi_0    pypi
trio-websocket            0.10.3                   pypi_0    pypi
twine                     4.0.2                    pypi_0    pypi
typer                     0.9.0              pyhd8ed1ab_0    conda-forge
typing-extensions         4.6.3                hd8ed1ab_0    conda-forge
typing_extensions         4.6.3              pyha770c72_0    conda-forge
typing_utils              0.1.0              pyhd8ed1ab_0    conda-forge
tzcode                    2023c                h1a8c8d9_0    conda-forge
tzdata                    2023c                h71feb2d_0    conda-forge
ukkonen                   1.0.1           py310h2887b22_3    conda-forge
unicodecsv                0.14.1                     py_1    conda-forge
unicodedata2              15.0.0          py310h8e9501a_0    conda-forge
unidecode                 1.3.6              pyhd8ed1ab_0    conda-forge
urllib3                   1.26.15            pyhd8ed1ab_0    conda-forge
validators                0.20.0             pyhd8ed1ab_0    conda-forge
vega_datasets             0.9.0              pyhd3deb0d_0    conda-forge
virtualenv                20.23.0            pyhd8ed1ab_0    conda-forge
vl-convert-python         0.12.0                   pypi_0    pypi
wcwidth                   0.2.6              pyhd8ed1ab_0    conda-forge
webencodings              0.5.1                      py_1    conda-forge
websocket-client          1.5.2              pyhd8ed1ab_0    conda-forge
wheel                     0.40.0             pyhd8ed1ab_0    conda-forge
wsproto                   1.2.0                    pypi_0    pypi
xerces-c                  3.2.4                h627aa08_1    conda-forge
xlrd                      2.0.1              pyhd8ed1ab_3    conda-forge
xlsxwriter                3.0.9              pyhd8ed1ab_0    conda-forge
xorg-libxau               1.0.11               hb547adb_0    conda-forge
xorg-libxdmcp             1.1.3                h27ca646_0    conda-forge
xyzservices               2023.5.0           pyhd8ed1ab_1    conda-forge
xz                        5.2.6                h57fd34a_0    conda-forge
yaml                      0.2.5                h3422bc3_2    conda-forge
yarl                      1.9.2           py310h2aa6e3c_0    conda-forge
zeromq                    4.3.4                hbdafb3b_1    conda-forge
zfp                       1.0.0                hb6e4faa_3    conda-forge
zict                      3.0.0              pyhd8ed1ab_0    conda-forge
zipp                      3.15.0             pyhd8ed1ab_0    conda-forge
zlib                      1.2.13               h03a7124_4    conda-forge
zlib-ng                   2.0.7                h1a8c8d9_0    conda-forge
zstd                      1.5.2                hf913c23_6    conda-forge
```

**Comment by wehuang16** [comment]
I used git clone to clone this powergenome repository to my local machine, then I used mamba (miniforge)  to create an environment with the environment.yml file ("mamba env create -f environment.yml"). I didn't use Anaconda because conda took too long to create the environment.

**Comment by gschivley** [comment]
Miniconda should also work. I think the trick is to make sure the Jupyter notebook is using the correct kernel.

Because I've always used conda, I install mamba in my `base` conda environment and then use `mamba env create -f environment.yml`. Maybe you'll have more luck with that?

**Comment by wehuang16** [comment]
Hey Greg, I appeared to make the downloading of powergenome working! While the command line "conda env create -f environment.yml" didn't work for me, I installed the conda packages (like geopandas, shapely) one at a time. All was successful but I failed to download pudl using conda: "conda install --channel=conda-forge catalystcoop.pudl=2022.11.30". I then tried doing "pip install catalystcoop.pudl" and then pudl was finally downloaded. 

Thank you for your help!
